Adonis-Cli Requires NPM Even When --yarn Flag Is On FIX

abrahamy opened the pull request:
This PR solves issue #39 . Instead of terminating with an error it gracefully terminates when npm is not installed.

The PR also added tests and travis/appveyor build scripts

thetutlage commented:
Ohh that's a nice catch, but the build is failing, would you mind looking into it

RomainLanz commented on src/Commands/Base.js:
```diff
@@ -43,9 +43,17 @@ class Base extends Command {
     let error = false
 
+    // retrieve npm version.
+    pify(exec)('npm -v').then(version => {
```
I'd prefer to use `yield pify(...)` inside a try/catch block instead of using Promise directly here.

abrahamy commented:
OK, I'll give it another shot and follow @RomainLanz's suggestion while at
it.

On Mon, May 29, 2017, 8:23 AM Romain Lanz <notifications@github.com> wrote:

> *@RomainLanz* commented on this pull request.
> ------------------------------
>
> In src/Commands/Base.js
> <https://github.com/adonisjs/adonis-cli/pull/40#discussion_r118879333>:
>
> >      let error = false
>
> +    // retrieve npm version.
> +    pify(exec)('npm -v').then(version => {
>
> I'd prefer to use yield pify(...) inside a try/catch block instead of
> using Promise directly here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/adonisjs/adonis-cli/pull/40#pullrequestreview-40711748>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH1MHyeH-kZ-CuGpWkD_a-uhN__oByBfks5r-nJpgaJpZM4NobaG>
> .
>

RomainLanz commented:
Also it would be maybe better to check first if `npm` is installed (by the same way we check if `yarn` is installed https://github.com/adonisjs/adonis-cli/blob/develop/src/Commands/Base.js#L102) instead of doing that while trying to fetch the version.

thetutlage commented:
I believe we are just doing too much here:

1. The simplest way to accept a parameter inside `checkRequirements` method on whether or not to validate npm version.
2. If user passes `--yarn` flag, then simply skip the npm check.

abrahamy commented:
I already did that in a branch of my fork. I'd merge it to my PR. Please
have a look at it and let me know what you think.

On Mon, May 29, 2017, 5:06 PM Harminder Virk <notifications@github.com>
wrote:

> I believe we are just doing too much here:
>
>    1. The simplest way to accept a parameter inside checkRequirements
>    method on whether or not to validate npm version.
>    2. If user passes --yarn flag, then simply skip the npm check.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/adonisjs/adonis-cli/pull/40#issuecomment-304693658>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH1MH7nx4I2vyuj7y2Gf1XBXRQci_1Nfks5r-u0TgaJpZM4NobaG>
> .
>

abrahamy commented:
@thetutlage @RomainLanz I have updated my PR and added tests and the build is no more failing. The reason the build was failing in the first instance was because `.travis.yml` and `appveyor.yml` build scripts were not in the repository. I have added these scripts and two tests to test for `--yarn` and `--npm` handling. Please kindly review and merge if all is as expected.

thetutlage commented:
@abrahamy Going to check it no

thetutlage commented:
Closed, since PR is for 3.x
